### ISSUE TITLE
xonsh $PATH does not include user env vars on Windows

### ISSUE THREAD

**Issue by gera2ld** [xonsh/xonsh#2320]
xonsh $PATH does not include user environment variables on Windows. As a result, some commands that can be found correctly in `cmd.exe` cannot be found in xonsh.

For example, my global PATHs are:
```python
['C:\\Windows\\System32', 'C:\\Windows', ...]
```
And my user PATHs are:
```python
['D:\\Programs\\nodejs', '...']
```

In xonsh, `$PATH` contains only the global PATHs, while `os.environ['PATH']` contains everything.

In Windows `cmd.exe`, `npm` works, but `xonsh` says `command not found`.
After typing `$PATH = os.environ['PATH'].split(';')`, it works.
BTW, if a command refers to a `.exe` file, it will be found even if its path not in `$PATH`, but not `.bat`.

So the question is, should I set `$PATH = os.environ['PATH'].split(';')` manually or do I miss anything?

**Comment by melund** [comment]
I can't reproduce this. Can you post the output of running `xonfig`

I get the same path no matter where I take it from. 
```
snail@sea ~\Documents\xonsh master
$ os_env_path = os.environ['PATH'].split(';')
snail@sea ~\Documents\xonsh master
$ cmd_path = $(cmd /c echo ! %PATH%).strip().split(';')
snail@sea ~\Documents\xonsh master
$ len(os_env_path), len(xonsh_path), len(cmd_path)
(44, 41, 44)
```

**Comment by melund** [comment]
I can't reproduce this. Can you post the output of running `xonfig` 

I get the same path no matter where I get it from: 
```
snail@sea ~\Documents\xonsh master
$ xonsh_path = set($PATH)
snail@sea ~\Documents\xonsh master
$ cmd_path = set($(cmd /c echo ! %PATH%).strip().split(';'))
snail@sea ~\Documents\xonsh master
$ os_env_path = set(os.environ['PATH'].split(';'))
snail@sea ~\Documents\xonsh master
$ len(xonsh_path), len(cmd_path), len(os_env_path)
(41, 41, 41)
```

**Comment by gera2ld** [comment]
![image](https://cloud.githubusercontent.com/assets/3139113/23942501/740c8d1c-09a7-11e7-8e4b-2b1d35c75f78.png)

By running `xonfig` I get:

| Name | Value |
|---|---|
| xonsh            | 0.5.8           |
| Python           | 3.6.0           |
| PLY              | 3.9             |
| have readline    | False           |
| prompt toolkit   | 1.0.13          |
| shell type       | prompt_toolkit  |
| pygments         | 2.2.0           |
| on posix         | False           |
| on linux         | False           |
| on darwin        | False           |
| on windows       | True            |
| on cygwin        | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |

FYI, I installed xonsh through miniconda with the commands [here](http://xon.sh/windows.html).

And something more, `$PATH` seems to read the raw PATH entries while `os.environ['PATH']` contains the expanded values.
![image](https://cloud.githubusercontent.com/assets/3139113/23943127/5ba9652c-09a9-11e7-9324-1056ef0df887.png)

**Comment by melund** [comment]
I think I got this figured. It is also a problem on my machine. I just didn't notice because the paths are not different. 

 It is caused by this PR here #2276.  That PR caused a lot of havoc in 0.57, which I thought we fixed by wrapping `os.environ` with a case-preserving/case-insensitive class #2298.  

So on windows the `PATH` variable is really called `Path`. Of course windows doesn't care about the case. But xonsh's `Env` class does. So it thinks `PATH` isn't available and thus falls back to a default value for PATH. On windows the default value is to load `PATH` from the registry: 
https://github.com/xonsh/xonsh/blob/cc0fdc5b2c75a72a41a90a40dcca9d36cef93f31/xonsh/environ.py#L797

I have put in a fix in #2325. @gera2ld  Can you try and see if it fixes it for you? Also, thanks a lot for spotting this....
